### PR TITLE
Add examples for random and manual partitioning

### PR DESCRIPTION
**Pull request by wvanbergen**
Towards addressing #323 and #302. @Shopify/kafka

### CODE REVIEW

File: partitioner_test.go [hunk 1]
```diff
@@ -120,3 +121,57 @@ func TestManualPartitioner(t *testing.T) {
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err := producer.Close(); err != nil {
+			log.Println("Failed to close producer:", err)
+		}
+	}()
+
+	msg := &ProducerMessage{Topic: "test", Partition: 6, Value: StringEncoder("test")}
```
Comment (eapache) [inline partitioner_test.go]: we should perhaps comment this line

File: partitioner_test.go [hunk 2]
```diff
@@ -120,3 +121,57 @@ func TestManualPartitioner(t *testing.T) {
 		}
 	}
 }
+
+// By default, Sarama uses the message's key to consistently assign a partition to
+// a message using hashing. If no key is not, a random partition will be chosen.
+// This example shows how you can partition messages randomly, even when a key is set
+func ExamplePartitioner_random() {
```
Comment (eapache) [inline partitioner_test.go]: not sure how useful this example is, it's literally just the one line config change...

**Comment by eapache** [comment]
just nits, no major objections

**Comment by eapache** [comment]
A more useful example might be writing your own partitioner, but I guess there's real live code (the existing implementations) for that

**Comment by wvanbergen** [comment]
It's only a one line change, but it doesn't necessarily make natural sense that this is part of the Config, instead of part of the ProducerMessage struct or SendMessage call

**Comment by eapache** [comment]
¯\_(ツ)_/¯

File: partitioner_test.go [hunk 3]
```diff
@@ -120,3 +121,62 @@ func TestManualPartitioner(t *testing.T) {
 		}
 	}
 }
+
+// By default, Sarama uses the message's key to consistently assign a partition to
+// a message using hashing. If no key is not, a random partition will be chosen.
```
Comment (eapache) [inline partitioner_test.go]: s/not/set

**Comment by eapache** [comment]
one typo, then :shipit:

**Pull request merged**